webpack.embedding-sdk-cli.config.js

```diff
@@ -0,0 +1,50 @@
+const path = require("path");
+const webpack = require("webpack");
+const TerserPlugin = require("terser-webpack-plugin");
+
+const SDK_CLI_DIST_PATH = path.join(__dirname, "/resources/embedding-sdk/dist");
+
+const SDK_CLI_PATH = path.join(
+  __dirname,
+  "/enterprise/frontend/src/embedding-sdk/cli",
+);
+
+const BABEL_CONFIG = {
+  cacheDirectory: process.env.BABEL_DISABLE_CACHE ? false : ".babel_cache",
+};
+
+module.exports = {
+  mode: "production",
+  entry: `${SDK_CLI_PATH}/cli.ts`,
```

Comment:
@deniskaber I was thinking if we can avoid webpack too. The main reason I went with webpack is to be able to bundle the CLI dependencies (e.g. `inquirer`, `commander`, `chalk`) into a single `./dist/cli.js` file, so it does not have to be in the `dependencies` of the package.json.

I was also a bit afraid of making silly mistakes without TypeScript, but we could use the [ts-check](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html) comment instead to enable type checks in a JavaScript file.

Reply:
We could also build it with `tsc` and then run the js files

Reply:
@npretto Let me try this with `tsc` instead of `webpack` today 👍🏻

Reply:
@deniskaber I tried with ES Modules and migrated to JavaScript in [this commit](https://github.com/metabase/metabase/pull/45784/commits/7d1b0651633bc5b4096715c3209be855b9163fb2) - it works fine in the Metabase directory, but when installing it into another directory it cannot find the dev dependencies.

The problem is that NPM/Yarn only installs the `dependencies` of an installed packaged, and not their `devDependencies`. We could add the CLI dependencies to the SDK's `dependencies` but that wouldn't be accurate.

Sadly, I think we'll need to stick with the webpack config, as the CLI dependencies (e.g. inquirer, commander, chalk) can be compiled into a single `cli.js` file without adding it to `dependencies`.

![CleanShot 2567-07-19 at 23 50 14@2x](https://github.com/user-attachments/assets/ece00a25-dfde-4a72-94ef-80235264a7db)

PS, @npretto I also tried with `tsc` but wasn't able to get it to work.

Reply:
> Why do you think it is an issue to have CLI dependencies in the SDK package.json?

It's because of how NPM [defines](https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file) `dependencies` and `devDependencies`.

> "dependencies": Packages required by your application in production.
> "devDependencies": Packages that are only needed for local development and testing.

The CLI portion is really only for local development, and not shipped into production - so adding CLI dependencies into `dependencies` for a SDK package would be misleading IMO.

As for the "unpacked size" that's fair. The CLI's current size is 132KB though, and a compiled `cli.js` is usually smaller than installing the dependencies separately.

Reply:
> size of all these bundled dependencies would be included into the sdk package size, even though it won't be used in the client app. 

While I agree it's not ideal, it seems it's pretty common for that number to be huge, here's nextjs with 86MB, or [aws-sdk](https://www.npmjs.com/package/aws-sdk) with 97MB

<img width="172" alt="image" src="https://github.com/user-attachments/assets/70adb0e4-bbe8-4ef5-8712-21c39cc6dadd">

I tried to look how nextjs does it, and it seems they have their own tool for this that they use in create-next-app: https://www.npmjs.com/package/@vercel/ncc
Should we give it a try? it should build a single js file with all the deps without much config needed

Reply:
@npretto sounds good, I'll be working on a follow up PR today to add automations (the second product doc), so it'll be worth trying out vercel/ncc and also [tsup](https://github.com/egoist/tsup). I'll re-test and merge this PR for now.